sky/serve/core.py

```diff
@@ -258,9 +259,23 @@ def up(
         else:
             lb_port = serve_utils.load_service_initialization_result(
                 lb_port_payload)
-            endpoint = backend_utils.get_endpoints(
-                controller_handle.cluster_name, lb_port,
-                skip_status_check=True).get(lb_port)
+
+            def _get_endpoint() -> str:
+                start_time = time.time()
+                while True:
+                    lb_endpoint = backend_utils.get_endpoints(
+                        controller_handle.cluster_name,
+                        lb_port,
+                        skip_status_check=True).get(lb_port)
+                    if lb_endpoint is not None:
+                        return lb_endpoint
+                    if (time.time() - start_time > serve_constants.
+                            SERVICE_CONTROLLER_INGRESS_TIMEOUT_SECONDS):
+                        raise RuntimeError(
+                            'Failed to get ingress endpoint for controller.')
+                    time.sleep(1)
+
+            endpoint = _get_endpoint()
```

Comment:
Should we instead add the retry in the underlying cloud implementation, as that is more like a cloud-specific issue, e.g. https://github.com/skypilot-org/skypilot/blob/master/sky/provision/kubernetes/network_utils.py#L258-L280